src/cider/nrepl/middleware/util/meta.clj

```diff
@@ -80,49 +93,30 @@
   (if-let [ns (find-ns ns)]
     (ns-aliases ns)))
 
-;; This reproduces the behavior of the canonical `clojure.repl/doc` using its
-;; internals, but returns the metadata rather than just printing. Oddly, the
-;; only place in the Clojure API that special form metadata is available *as
-;; data* is a private function. Lame. Just call through the var.
```

Comment:
You had some great rationale on what's true a special form, so you might put it here for future reference.

Reply:
Ratinale added!